src/sql/workbench/contrib/notebook/browser/outputs/notebookMarkdown.ts

```diff
@@ -111,7 +111,11 @@ export class NotebookMarkdownRenderer {
 				text = removeMarkdownEscapes(text);
 			}
 			title = removeMarkdownEscapes(title);
-			href = removeMarkdownEscapes(href);
+			// only remove escaped characters if it's a hyperlink, filepath usually can start with .{}_
+			// and the below function escapes them if it encounters in the path.
+			if (!path.isAbsolute(href)) {
```

Comment:
This is still going to break if you have a relative link to a folder with a . in it though, right? Do we ever even escape these links? It would seem like the correct thing to do here is ensure that the markdown links are properly escaped before this is called.

Reply:
relative links are resolved before coming to this check.

Reply:
Then what's the point of the isAbsolute check? What paths can make it here that aren't absolute and need to be unescaped?

Reply:
http links

Reply:
In path.test.ts we can also see that 'http://foo.com/bar' is a relative path

Reply:
now that's confusing, they even have an absolute path like z:/foo/bar.txt in relative path.

Reply:
I believe that's for `path.posix.isAbsolute`, which will return a relative path.